dimod/reference/composites/fixedvariable.py

```diff
@@ -0,0 +1,103 @@
+import dimod
+from dimod.roof_duality import fix_variables
+from dimod.response import SampleSet
+import numpy as np
+
+__all__ = ['FixedVariableComposite']
+
+class FixedVariableComposite(dimod.ComposedSampler):
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+
+        return self._children
+
+    @property
+    def parameters(self):
+
+        return self.child.parameters.copy()
+
+    @property
+    def properties(self):
+
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, source_bqm, fixed_variables=None,
+               **parameters):
+
+        child = self.child
+        if fixed_variables is None:
+            fixed_variables = fix_variables(source_bqm)
+        bqm = source_bqm.copy()
+
+        for v, val in fixed_variables.items():
```

Comment:
We have a new method `fix_variables` that you can use.

Reply:
ok i'll change this